### PR TITLE
[release/6.0.4xx]conditions syntax test

### PR DESCRIPTION
**Pull request by vlada-shubina**
### Problem
The PR showcases use of different boolean expressions with available evaluators (CPP, CPP2, VB, MSBUILD).

### CODE REVIEW

File: test/dotnet-new3.UnitTests/DotnetNewInstantiate.cs [hunk 1]
```diff
@@ -591,5 +594,37 @@ public void DoesNotReportErrorOnDefaultUpdateCheckOfLocalPackageDuringInstantiat
                 .And.NotHaveStdErr()
                 .And.HaveStdOutContaining("File actions would have been taken:");
         }
+
+        [Theory]
+        [InlineData("no options", new[] { "TestAssets.TemplateWithBooleanParameters" })]
+        [InlineData("options without value", new[] { "TestAssets.TemplateWithBooleanParameters", "--A", "--B", "--C", "--D" })]
+        [InlineData("option equals to false", new[] { "TestAssets.TemplateWithBooleanParameters", "--A", "false", "--B", "false", "--C", "false", "--D", "false" })]
+        [InlineData("option equals to true", new[] { "TestAssets.TemplateWithBooleanParameters", "--A", "true", "--B", "true", "--C", "true", "--D", "true" })]
```
Comment (JanKrivanek) [inline test/dotnet-new3.UnitTests/DotnetNewInstantiate.cs:602]: Should we add testcases for unspecified value and missing default value?

**Review by JanKrivanek — APPROVED** [review]
Helpful overview of current functionality!
I'm wondering if we should test the missing default value as well.

File: test/Microsoft.TemplateEngine.TestTemplates/test_templates/TemplateWithBooleanParameters/.template.config/template.json [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  "generatorVersions": "[1.0.0.0-*)",
+  "groupIdentity": "TestAssets.TemplateWithBooleanParameters",
+  "precedence": "100",
+  "identity": "TestAssets.TemplateWithBooleanParameters",
+  "shortName": "TestAssets.TemplateWithBooleanParameters",
+  "symbols": {
+    "A": {
+      "type": "parameter",
+      "dataType": "bool"
+    },
```
Comment (vlada-shubina) [inline test/Microsoft.TemplateEngine.TestTemplates/test_templates/TemplateWithBooleanParameters/.template.config/template.json:14]: link

File: test/dotnet-new3.UnitTests/DotnetNewInstantiate.cs (hunk 1 above)
Comment (vlada-shubina) [reply test/dotnet-new3.UnitTests/DotnetNewInstantiate.cs:602]: This should be covered by parameter A: https://github.com/dotnet/templating/pull/5728/files#r1053118152 or do you mean something else here?

**Comment by vlada-shubina** [comment]
backport to release/7.0.2xx

**Pull request merged**